tutor/templates/apps/openedx/settings/partials/common_all.py

```diff
@@ -6,17 +6,16 @@
 
 # Mongodb connection parameters: simply modify `mongodb_parameters` to affect all connections to MongoDb.
 mongodb_parameters = {
+    "db": "{{ MONGODB_DATABASE }}",
     "host": "{{ MONGODB_HOST }}",
     "port": {{ MONGODB_PORT }},
-    {% if MONGODB_USERNAME and MONGODB_PASSWORD %}
-    "user": "{{ MONGODB_USERNAME }}",
-    "password": "{{ MONGODB_PASSWORD }}",
-    {% else %}
-    "user": None,
-    "password": None,
-    {% endif %}
-    "db": "{{ MONGODB_DATABASE }}",
-    "replicaSet": None,
+    "user": {% if MONGODB_USERNAME %}"{{ MONGODB_USERNAME }}"{% else %}None{% endif %},
+    "password": {% if MONGODB_PASSWORD %}"{{ MONGODB_PASSWORD }}"{% else %}None{% endif %},
+    # Connection/Authentication
+    "ssl": {{ MONGODB_USE_SSL }},
+    "authSource": "{{ MONGODB_AUTH_SOURCE }}",
+    "replicaSet": {% if MONGODB_REPLICA_SET %}"{{ MONGODB_REPLICA_SET }}"{% else %}None{% endif %},
+    "authMechanism": "{{ MONGODB_AUTH_MECHANISM }}",
```

Comment:
We shouldn't set the `authMechanism` key unless it's been defined by the user. The library doesn't allow setting it without providing authentication credentials. Note that you can't make it an empty string either as it'll complain that you need to set it to a valid value.

We should do something like this:

```python
{% if MONGODB_AUTH_MECHANISM %}
mongodb_parameters["authMechanism"] = "{{ MONGODB_AUTH_MECHANISM }}"
{% endif %}
```

```python
  Applying split_modulestore_django.0002_data_migration...Traceback (most recent call last):
  File "./manage.py", line 106, in <module>
    execute_from_command_line([sys.argv[0]] + django_args)
  File "/openedx/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/openedx/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/openedx/venv/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/openedx/venv/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/openedx/venv/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/openedx/venv/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/openedx/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/openedx/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/openedx/venv/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/openedx/venv/lib/python3.8/site-packages/django/db/migrations/migration.py", line 123, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/openedx/venv/lib/python3.8/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/openedx/edx-platform/common/djangoapps/split_modulestore_django/migrations/0002_data_migration.py", line 24, in forwards_func
    split_modulestore = modulestore()._get_modulestore_by_type(ModuleStoreEnum.Type.split)
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/modulestore/django.py", line 319, in modulestore
    contentstore(),
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/contentstore/django.py", line 29, in contentstore
    _CONTENTSTORE[name] = class_(**options)
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/contentstore/mongo.py", line 44, in __init__
    mongo_db = connect_to_mongodb(
  File "/openedx/edx-platform/common/lib/xmodule/xmodule/mongo_utils.py", line 55, in connect_to_mongodb
    pymongo.MongoClient(
  File "/openedx/venv/lib/python3.8/site-packages/pymongo/mongo_client.py", line 718, in __init__
    self.__options = options = ClientOptions(
  File "/openedx/venv/lib/python3.8/site-packages/pymongo/client_options.py", line 156, in __init__
    self.__credentials = _parse_credentials(
  File "/openedx/venv/lib/python3.8/site-packages/pymongo/client_options.py", line 38, in _parse_credentials
    return _build_credentials_tuple(
  File "/openedx/venv/lib/python3.8/site-packages/pymongo/auth.py", line 113, in _build_credentials_tuple
    raise ConfigurationError("%s requires a username." % (mech,))
pymongo.errors.ConfigurationError: SCRAM-SHA-256 requires a username.
```

Reply:
:ok_hand: